### ISSUE TITLE
The GPU utilization graph drawing is broken (being updated incorrectly)

### ISSUE THREAD

**Issue by ivanp7** [Syllo/nvtop#200]
I've installed nvtop 3.0.1 on Gentoo using ebuild from the GURU repository, and the graph drawing is broken like this

![image](https://user-images.githubusercontent.com/3432246/229540141-22249b0e-01b5-4bd3-a2f2-58ca8d82b7b1.png)

On Artix Linux there wasn't such a problem.

**Comment by ivanp7** [comment]
ldd reports that the binary is linked to the wide ncurses
![image](https://user-images.githubusercontent.com/3432246/229600594-bcefdd2d-91cb-49f2-ad85-d2f5294b927c.png)

**Comment by WhiskeyTuesday** [comment]
I have the same issue on pop_os/ubuntu 22.04

![image](https://user-images.githubusercontent.com/3824702/231049187-bd0716eb-f41e-44d8-a50e-70e0b3a3f7f9.png)

**Comment by WhiskeyTuesday** [comment]
Actually it may be a related but not exactly the same issue. Both in tmux in alacritty and in gnome-term (in tmux or not) the top bar disappears a second or so after launch and the longer it runs the more screwy it becomes

![image](https://user-images.githubusercontent.com/3824702/231049510-5b9f5ce7-e37e-4202-aa0b-cd59062300f4.png)
![image](https://user-images.githubusercontent.com/3824702/231049537-c87a28ad-3ea1-4cb9-ae3c-d683eb7c5c9d.png)

**Comment by zhuyifei1999** [comment]
FWIW, being on Gentoo by itself doesn't cause what you see, nor does using unicode ncurses:
![Screenshot_2023-04-10_20-50-39](https://user-images.githubusercontent.com/2096920/231052013-71ebf363-b42e-475b-b50b-6f479c4d2832.png)

In fact, it's unicode ncurses is preferred: https://github.com/Syllo/nvtop/blob/37f1b16bebbe8bb08da3d1f74a9cdbdbd7e4316a/CMakeLists.txt#L27-L35

**Comment by Syllo** [comment]
Hello,
Using ncurses**w** should be the default, so I don't think that it comes from a ncurses vs ncursesw issue.

This could be a bug in the drawing routine.
@ivanp7 does it happen with any terminal size?
Can you give me the terminal size used when the issue happens?

**Comment by Syllo** [comment]
Could be related to #192

**Comment by ivanp7** [comment]
@Syllo Indeed, it is related to #192. I have tried `tabs -8` and it worked flawlessly (my terminal uses 4 chars for tabs by default). However, I use the same terminal with the same tab setting on Artix Linux, and nvtop does not reproduce the bug there. As for the terminal size, it happens with any size.

**Comment by Syllo** [comment]
Could it be a terminal emulator issue? Is it the same on both systems?

**Comment by ivanp7** [comment]
It is compiled from the same source and uses the same configuration. Generally, it does work without glitches in ncurses applications. It does not cause the nvtop bug on Artix Linux (where tab is 4 chars).

**Comment by ivanp7** [comment]
I have just noticed that nvtop graph still glitches even with `tabs -8`, just less often with less artifacts.

**Comment by ivanp7** [comment]
I see no artifacts when nvtop is running in tmux.

**Comment by WhiskeyTuesday** [comment]
This only seems to happen to me (Pop_OS 22.04, nvtop 1.2.2) if I have sort set to GPU or GPU MEM, and seems to be triggered by nvtop trying to print "Comparing GPU rates" somewhere on screen and getting layout messed up as a result.

If I sort by PID, CPU, or anything else I've tried so far that message is never displayed, and the rendering never gets corrupted.

Happens both in and out of tmux, and in all terminal emulators that I have installed (gnome-term, alacritty, wezterm, cool-retro-term)

**Comment by WhiskeyTuesday** [comment]
I have tried on several terminals both in and out of tmux. The messed up plotting seems to happen at the same time as the vertical shift that is caused by the debug message. When it happens the top line of information at the top of the display cuts off, and the various other issues occur at the same time, including shifting the "tops"  of each spike on the graphs up by what looks to me like one line, and from then on they persist where they are with the correct graph still being drawn underneath (and continuing to scroll to the left).

![image](https://user-images.githubusercontent.com/3824702/231878142-c9f91cf8-3edb-4336-a96c-70bd8d05a3aa.png)
![Screenshot from 2023-04-13 13-38-29](https://user-images.githubusercontent.com/3824702/231878261-115112ec-4301-4ae7-82c6-d45332dc5f0f.png)

**Comment by WhiskeyTuesday** [comment]
Hitting F6 to bring up the sort menu, or Esc if it is already open, either of which trigger a full page refresh I suspect, fixes the problem for a few ticks until the debug message is printed again.

**Comment by Syllo** [comment]
@WhiskeyTuesday the debug print has been [removed in a patch already](https://github.com/Syllo/nvtop/commit/13e684ed97f5837a5e50e2fe9c712890560c9f1b) and made it in nvtop versions >= 2. So I would advise you to use a more recent version of nvtop (e.g., use the AppImage).

**Comment by Syllo** [comment]
Are you guys still experiencing this issue?

**Comment by ivanp7** [comment]
Unfortunately, yes
